File: NSURLExample/NSURLExampleInObjc/NSURLExampleInObjc/ViewController.m
//
//  ViewController.m
//  NSURLExampleInObjc
//
//  Created by ShannonChen on 2017/5/19.
//  Copyright © 2017年 ShannonChen. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    
    // Absolute URL and relative URL
    NSURL *baseURL = [NSURL URLWithString:@"file:///user/Documents"];
    //    NSURL *baseURL = [NSURL URLWithString:@"file:///user/Documents/"];  // 最后的斜杠“/”会影响后面使用 -URLWithString:relativeToURL: 方法拼接时的结果
    NSURL *anURL = [NSURL URLWithString:@"./Library/Cache" relativeToURL:baseURL];
    
    NSLog(@"%@", anURL.absoluteString);  // file:///user/Library/Cache
    NSLog(@"%s", anURL.fileSystemRepresentation); // /user/Library/Cache
    NSLog(@"%@", anURL.standardizedURL.absoluteString);
    NSLog(@"%@", anURL.filePathURL);
    NSLog(@"%@", anURL.fileReferenceURL);
    NSString *str =nil;
    [str stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
    NSArray *components = @[@"user", @"doc", @"music"];
    NSURL *fileURL = [NSURL fileURLWithPathComponents:components];
}




@end

File: NSURLExample/NSURLExampleInSwift/NSURLExampleInSwift/ViewController.swift
//
//  ViewController.swift
//  NSURLExampleInSwift
//
//  Created by ShannonChen on 2017/5/19.
//  Copyright © 2017年 ShannonChen. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Absolute URL and relative URL
        let baseURL = URL.init(string: "file:///user/Documents")
        let anURL = NSURL.init(string:"./Library/Cache", relativeTo: baseURL)
        
        if let absoluteString = anURL?.absoluteString {
            print(absoluteString)  // file:///user/Library/Cache
        }
        
        
        // Nested Types
    }

}

